src/RobustLeastSquares.jl

```diff
@@ -1,5 +1,307 @@
 module RobustLeastSquares
 
+using StatsBase
+using IterativeSolvers
+using Logging
+
+export reweighted_least_squares, mestimator_mad
+export MEstimator, MEstimatorConvex, MEstimatorNonConvex
+export L2Estimator, L1Estimator, L1L2Estimator, HuberEstimator, FairEstimator, CauchyEstimator, SubsetEstimator, SubsetEstimatorConvex, DualEstimator, DualEstimatorConvex
+
+
+
+using PyCall
+@pyimport pyamg
+@pyimport scipy.sparse as scipy_sparse
+
+# Helper function for python matrices
+function py_csc(A::SparseMatrixCSC)
+    # create an empty sparse matrix in Python
+    Apy = scipy_sparse.csc_matrix(size(A))
+    # write the values
+    Apy[:data] = copy(A.nzval)
+    # write the indices
+    Apy[:indices] = A.rowval - 1
+    Apy[:indptr] = A.colptr - 1
+    return Apy
+end
+py_csr(A::SparseMatrixCSC) = py_csc(A)[:tocsr]()
+
+
+# Weight functions
+# ----------------
+# A good overview of these can be found at:
+# http://research.microsoft.com/en-us/um/people/zhang/INRIA/Publis/Tutorial-Estim/node24.html
+
+# Standard L2 estimator
+"""
+An m-estimator is a cost/loss function used in modified (weighted) least squares
+problems of the form:
+    min ∑ᵢ ρ(rᵢ)
+"""
+abstract MEstimator
+abstract MEstimatorConvex <: MEstimator
+abstract MEstimatorNonConvex <: MEstimator
+
+# Default implementations from definintion of cost-function rho
+"The cost (a.k.a. loss) function ρ for the M-estimator"
+function m_estimator_rho end
+"The derivative of the cost (a.k.a. loss) function ψ for the M-estimator"
+function m_estimator_psi end
+# TODO m_estimator_psi(r,est::MEstimator) = first derivative of m_estimator_rho(r,est) w.r.t. r
+"The weight function, w, for the M-estimator, to be used for modifying least-square problems"
+m_estimator_weight(r,est::MEstimator) = m_estimator_psi(r,est) ./ r
+"The square root of the weight function, sqrt(w), for the M-estimator, to be used for modifying the normal equations of a least-squares problem"
```

Comment:
This is nitpicking, but could you line wrap the comments (it's impossible to see them here in the interface)